app/javascript/packs/reducers/buildings.js

```diff
@@ -78,14 +85,106 @@ function saveBuilding(state, action) {
   };
 }
 
+  /////////////////////////////////////////////////////////////////
+ // ANSWERS //////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////
+
+/**
+ * Updates the answer in store with the data that is currently being sent through a fetch request.
+ */
+function beforeFetchAnswer(state, action) {
+  const buildingId = action.buildingId;
+  const answer = action.answer;
+
+  return {
+    ...state,
+    [buildingId]: {
+      ...state[buildingId],
+      answers: {
+        ...state[buildingId].answers,
+        [answer.question_id]: {
+          ...answer,
+          saved: false,
+          fetching: true,
+          buildingId: buildingId,
+        }
+      },
+    }
+  };
+}
+
+/**
+ * Updates the answer in store with the response received from the fetch request.
+ */
+function updateAnswer(state, action) {
+  const status = action.status;
+  const answer = action.response;
+  const buildingId = answer.building_id;
+
+  if (status === FETCH_SUCCESS) {
+    return {
+      ...state,
+      [buildingId]: {
+        ...state[buildingId],
+        answers: {
+          ...state[buildingId].answers,
+          [answer.question_id]: {
+            ...answer,
+            saved: true,
+            fetching: false,
+            error: false
+          }
+        }
+      }
+    }
+  }
+
+  if (status === FETCH_FAILURE) {
+    const errorMessage = action.response;
+    return {
+      ...state,
+      [buildingId]: {
+        ...state[buildingId],
+        answers:{
+          ...state[buildingId].answers,
+          [answer.id]: {
+            ...state[buildingId].answers[answer.id],
+            error: errorMessage,
+            fetching: false
+          }
+        }
+      }
+    }
+  }
+
+  return {
+    ...state,
+    [buildingId]: {
+      ...state[buildingId],
+      answers: {
+        ...state[buildingId]['answers'],
+        [answer.id]: {
+          ...state[buildingId]['answers'][answer.id],
+          fetching: true,
+        }
+      }
+    }
+  }
+}
+
 export default function buildings(state = {}, action) {
   if (!action) return state;
   switch (action.type) {
+    // Buildings
     case ADD_BUILDING: return addBuilding(state, action);
     case EDIT_BUILDING: return editBuilding(state, action);
     case REMOVE_BUILDING: return removeBuilding(state, action);
     case CREATE_BUILDING: return saveBuilding(state, action);
     case UPDATE_BUILDING: return saveBuilding(state, action);
+
+    // Answers
+    case FETCHING_ANSWER: return beforeFetchAnswer(state, action);
```

Comment:
One thing we can do to improve this structure is export answers into a separate reducer and respond to that in these answers cases with:

```javascript
default: return {...state, answers: beforeFetchAnswer(state.answers, action) }
```
Let me know what you think!

Reply:
So would the answers reducer get the building object as state instead of the answer object? I thought you meant that the default case would go in the buildings reducer, and it would say something like:

`default: return {...state, answers: answers(state.answers, action) }`

(where `answers()` is the answers reducer itself)

Reply:
Yes, this is what I meant!